llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp

```diff
@@ -400,28 +400,27 @@ void AMDGPUAsmPrinter::emitCommonFunctionComments(
                               false);
 }
 
-std::string AMDGPUAsmPrinter::getMCExprStr(const MCExpr *Value) {
-  std::string Str;
-  raw_string_ostream OSS(Str);
+StringRef AMDGPUAsmPrinter::getMCExprStr(const MCExpr *Value) {
```

Comment:
You can't return this StringRef references to the local SmallString. You have to return the SmallString